security/pkg/nodeagent/cache/helper.go

```diff
@@ -21,13 +21,15 @@ import (
 	"strings"
 
 	"google.golang.org/grpc/codes"
+
+	credPlugin "istio.io/istio/security/pkg/credentialfetcher/plugin"
 )
 
-func constructCSRHostName(trustDomain, token string) (string, error) {
+func constructCSRHostName(platform, trustDomain, token string) (string, error) {
```

Comment:
Do we even need this ? Istiod can set it when it generates the cert - from the info in the cert.

Reply:
The code has a lot of complexity - agent is trying to guess, and server needs to verify. 

In reality - the SAN is based on info that the server has from the JWTs and config.